libvast/src/detail/sigma.cpp

```diff
@@ -0,0 +1,369 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/detail/sigma.hpp"
+
+#include "vast/concept/parseable/core.hpp"
+#include "vast/concept/parseable/string.hpp"
+#include "vast/detail/base64.hpp"
+#include "vast/detail/string.hpp"
+#include "vast/error.hpp"
+#include "vast/expression_visitors.hpp"
+
+#include <caf/optional.hpp>
+
+#include <map>
+#include <regex>
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace vast::detail::sigma {
+
+// TODO: A lot of code in here is directly copied from
+// src/concept/parseable/expression.cpp. We should factor the implementation in
+// the future.
```

Comment:
Please write a story for this, otherwise it won't ever get done.

Also, a corresponding comment in the expression parser is needed to make sure we don't make changes to it that aren't mirrored here.

Reply:
Done.